pkg/network/protocols/http2/telemetry.go

```diff
@@ -57,18 +60,45 @@ func newHTTP2KernelTelemetry() *kernelTelemetry {
 
 // update updates the kernel metrics with the given telemetry.
 func (t *kernelTelemetry) update(tel *HTTP2Telemetry) {
-	t.http2requests.Add(int64(tel.Request_seen))
-	t.http2responses.Add(int64(tel.Response_seen))
-	t.endOfStream.Add(int64(tel.End_of_stream))
-	t.endOfStreamRST.Add(int64(tel.End_of_stream_rst))
-	t.pathExceedsFrame.Add(int64(tel.Path_exceeds_frame))
-	t.exceedingMaxInterestingFrames.Add(int64(tel.Exceeding_max_interesting_frames))
-	t.exceedingMaxFramesToFilter.Add(int64(tel.Exceeding_max_frames_to_filter))
+	// We should only add the delta between the current eBPF map state and the last seen eBPF map state
+	telemetryDelta := tel.Sub(t.telemetryLastState)
+	t.http2requests.Add(int64(telemetryDelta.Request_seen))
+	t.http2responses.Add(int64(telemetryDelta.Response_seen))
+	t.endOfStream.Add(int64(telemetryDelta.End_of_stream))
+	t.endOfStreamRST.Add(int64(telemetryDelta.End_of_stream_rst))
+	t.pathExceedsFrame.Add(int64(telemetryDelta.Path_exceeds_frame))
+	t.exceedingMaxInterestingFrames.Add(int64(telemetryDelta.Exceeding_max_interesting_frames))
+	t.exceedingMaxFramesToFilter.Add(int64(telemetryDelta.Exceeding_max_frames_to_filter))
 	for bucketIndex := range t.pathSizeBucket {
-		t.pathSizeBucket[bucketIndex].Add(int64(tel.Path_size_bucket[bucketIndex]))
+		t.pathSizeBucket[bucketIndex].Add(int64(telemetryDelta.Path_size_bucket[bucketIndex]))
 	}
+	// Create a deep copy of the 'tel' parameter to prevent changes from the outer scope affecting the last state
+	t.telemetryLastState = *tel
```

Comment:
why not use a value instead of a pointer everywhere?

IMO `func (t *kernelTelemetry) update(tel *HTTP2Telemetry)` should become:

``` 
func (t *kernelTelemetry) update(tel HTTP2Telemetry)
``` 
Given that this method doesn't mutate the struct and would also update
```
func (t *HTTP2Telemetry) Sub(other HTTP2Telemetry) *HTTP2Telemetry
```
to

```
func (t *HTTP2Telemetry) Sub(other HTTP2Telemetry) HTTP2Telemetry
```

Reply:
That's a good point, but I prefer not to change the `HTTP2Telemetry` variable type in `updateKernelTelemetry` as it may introduce some artifacts with `Lookup` function, and require further testing. We can definitely do it in another PR